test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs

```diff
@@ -469,6 +469,244 @@ long GetAggregatedValue(Metric metric)
             }
         }
 
+        [Fact]
+        public void DuplicateInstrumentRegistration_WithViews_TwoIdenticalInstruments_TwoViews_SameRename()
+        {
+            var exportedItems = new List<Metric>();
+
+            using var meter = new Meter($"{Utils.GetCurrentMethodName()}");
+            var meterProviderBuilder = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
+                .AddView("name", "newname")
+                .AddView("name", "newname")
```

Comment:
Would it be a good idea to throw here (e.g. it's possible to detect that two `View`s are using the same `newname` before we see any instruments).

Reply:
Yes, agreed. @cijothomas and I have discussed this a bit. There are definitely some improvements we can make at view registration time and fail-fast when we know for certain a view will cause conflicts.

It's namely our `AddView(Func<Instrument, MetricStreamConfig>)` overload where we won't be able to detect a conflict at registration time. I think I will change this test to use that overload instead to achieve a similar result.

I'd like to consider cases where we can fail fast at registration in a different PR.